src/BinSkim.Rules/PERules/BA2025.EnableIntegrityCheck.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Reflection.PortableExecutable;
+
+using Microsoft.CodeAnalysis.BinaryParsers;
+using Microsoft.CodeAnalysis.IL.Sdk;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.IL.Rules
+{
+    [Export(typeof(Skimmer<BinaryAnalyzerContext>)), Export(typeof(ReportingDescriptor))]
+    public class EnableIntegrityCheck : PEBinarySkimmerBase
+    {
+        /// <summary>
+        /// https://docs.microsoft.com/windows/win32/debug/pe-format#dll-characteristics
+        /// </summary>
+        private const string ImageDllCharacteristicsForceIntegrity = "128";
+
+        /// <summary>
+        /// BA2025
+        /// </summary>
+        public override string Id => RuleIds.EnableIntegrityCheck;
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.BA2025_EnableIntegrityCheck_Description };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.BA2025_Note_IntegrityCheckShouldBeEnabled),
+            nameof(RuleResources.BA2025_Pass),
+            nameof(RuleResources.NotApplicable_InvalidMetadata)
+        };
+
+        public override void Analyze(BinaryAnalyzerContext context)
+        {
+            PEBinary target = context.PEBinary();
+            PEHeader peHeader = target.PE.PEHeaders.PEHeader;
+
+            DllCharacteristics dllCharacterstics = peHeader.DllCharacteristics;
+            if (!dllCharacterstics.HasFlag((DllCharacteristics)Enum.Parse(typeof(DllCharacteristics), ImageDllCharacteristicsForceIntegrity)))
```

Comment:
>Enum.Parse(typeof(DllCharacteristics), ImageDllCharacteristicsForceIntegrity)) [](start = 63, length = 78)

Just embed a value of 0x80 here and cast it to DllCharacteristics. There's no good reason to parse a string representation to it.